### PR TITLE
Zero out stencil bit after they are not needed anymore

### PR DESCRIPTION
**Pull request by FrancescoC-unity**
### Purpose of this PR
Clean the stencil bits when we don't need them anymore so to facilitate the reusage of them. 

---
### Testing status
**Katana Tests**: https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?automation-tools_branch=master&ScriptableRenderLoop_branch=HDRP/free-stencil-bits-when-unneeded-anymore&unity_branch=trunk

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo
- [ ] C# and shader warnings (supress shader cache to see them)
- Other: 

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**:  Low

**Halo Effect**:  Low

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/PostProcessing/PostProcessSystem.cs [hunk 1]
```diff
@@ -21,13 +21,8 @@ private enum SMAAStage
...
         {
-            // We don't have bits left to use nor we want to reserve them for SMAA
-            // so we use the same mask as DoesntReceiveSSR as this is likely to not contain
-            // a lot of objects already marked. Worst case scenario, we are running SMAA on more pixel
-            // than needed, but it should still be faster than clearing the stencil.
-            // TODO: We can probably free up some bits while writing others, by leveraging the limited lifetime
-            // of some of the bits stored. But this require a change that is orthogonal to SMAA.
-            SMAABit = HDRenderPipeline.StencilBitMask.DoesntReceiveSSR
+            // DistortionVectors bit is not needed anymore at this stage.
+            SMAABit = HDRenderPipeline.StencilBitMask.DistortionVectors
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/PostProcessing/PostProcessSystem.cs]: Please, use an explicit name HDRenderPipeline.StencilBitMask.SMAABit  but use the same value as for DistortionVectors instead. It is very error prone to do this here where people editing HDRenderPipeline.StencilBitMask can't see it.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/MotionVectors/CameraMotionVectors.shader [hunk 2]
```diff
@@ -74,10 +74,11 @@ Shader "Hidden/HDRP/CameraMotionVectors"
             // We will perform camera motion velocity only where there is no object velocity
             Stencil
             {
+                WriteMask 128
                 ReadMask 128
                 Ref  128 // StencilBitMask.ObjectVelocity
                 Comp NotEqual
-                Pass Keep
+                Fail Zero   // We won't need the bit anymore.
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/RenderPass/MotionVectors/CameraMotionVectors.shader]: Let's discuss

**Comment by FrancescoC-unity** [comment]
Rekicking visual tests

**Comment by FrancescoC-unity** [comment]
And new tests are green too

**Pull request merged**